### ISSUE TITLE
Missing version on `metadata.json` makes it failed to start

### ISSUE THREAD

**Issue by alex285** [jderose9/dash-to-panel#319]
hello, i pulled from the master and i noticed that the extension was failing to start. the problem is on `metadata.json` file that the `version` property was empty, i guess after https://github.com/jderose9/dash-to-panel/commit/9223a67ebf6d22357fe7fb32e68555d5778240b1

```
{
"extension-id": "dash-to-panel",
"uuid": "dash-to-panel@jderose9.github.com",
"name": "Dash to Panel",
"description": "An icon taskbar for the Gnome Shell. This extension moves the dash into the gnome main panel so that the application launchers and system tray are combined into a single panel, similar to that found in KDE Plasma and Windows 7+. A separate dock is no longer needed for easy access to running and favorited applications.\n\nFor a more traditional experience, you may also want to use Tweak Tool to enable Windows > Titlebar Buttons > Minimize & Maximize.",
"shell-version": [ "3.18", "3.20", "3.22", "3.24", "3.26" ],
"url": "https://github.com/jderose9/dash-to-panel",
"gettext-domain": "dash-to-panel",
"version": ,
"commit": "9223a67ebf6d22357fe7fb32e68555d5778240b1"
}
```

**Comment by jderose9** [comment]
Ok, thanks for reporting! I'm guessing you did a shallow clone or downloaded the zip from github? Is that right? I'm pulling the latest tags from the git repo and didn't account for that. This was to stop the chrome-gnome-shell update notifier from constantly nagging that a newer version was available on e.g.o. (when it wasn't, the metadata just didn't have the current version).

I can get the tags from github during make but it wouldn't necessarily correlate to the version of the code you have on your machine. I could also default it to a really high version number, but you would never get an update notification if you had installed this way, even if a new release came out. Maybe it would be best just to commit the current version each time I submit to e.g.o...

**Comment by alex285** [comment]
@jderose9  yes that is correct. im using a shallow clone, and i only update this extension from Github, rather EGO!

it seems more correct to submit a normal version on EGO, but personally i dont mind of this issue. i just open a bug report to let you know ;)

**Comment by zagortenay333** [comment]
Why do you guys use a version in the metadata file at all? Set it to -1 and no more problems with those pesky notifications.

**Comment by jderose9** [comment]
> Why do you guys use a version in the metadata file at all? Set it to -1 and no more problems with those pesky notifications.

That actually causes the extension to always be marked as outdated. The update check api just does a integer comparison on the version number.

https://github.com/GNOME/extensions-web/blob/d01ebe0cb04184df7e68717bfacf0ae71e837f97/sweettooth/extensions/views.py#L142

Also, the hope with adding the closest version number is that the user could get a notification later when a new version was released on e.g.o.

**Comment by zagortenay333** [comment]
> > Why do you guys use a version in the metadata file at all? Set it to -1 and no more problems with those pesky notifications.
> 
> That actually causes the extension to always be marked as outdated. The update check api just does a integer comparison on the version number.

Oh, I thought that was a way to tell them not bother with the versions...

How did it work for me this whole time though?
I have it set to -1 in my extension and never get notifs that I should upgrade. When I go the the extensions site, the green upgrade button is there _(because of the -1 in the local version)_, but it never shows me notifications.
Do the notifications start appearing only when you install it via the extension site?

**Comment by jderose9** [comment]
The notifications show up if you have the chrome-gnome-shell package installed with update notifications enabled in the settings.

**Comment by zagortenay333** [comment]
I'm getting notifs for dash-to-dock but not for my extension.
